### DISCUSSION TITLE
How to disable temporary insertion by `menu-complete`?

### DISCUSSION THREAD

**Discussion by DUOLabs333** [akinomyoga/ble.sh#252] [Q&A] [answered]
That is, when TAB is pressed, don't modify the prompt until a suggestion is selected?

**Comment by akinomyoga** [comment]
Sorry for the delay. There was no way to disable the auto completion after the TAB completion. I have decided to add an option for that in commit 1478a049. Now you can update ble.sh by running `ble-update` and write the settings as follows:

<details><summary>(outdated)</summary>

```bash
# blerc

bleopt complete_auto_after_complete=
```
</details>

**Edit (2022-12-09): I have changed the option name and the value:**

```bash
# blerc

bleopt complete_auto_complete_opts=suppress-after-complete
```



Note that, if you are using ble-0.3, you need to switch to 0.4.

@DUOLabs333 Could you check if it is what you want? Thank you!

**Reply by DUOLabs333** [reply]
I don't know if I'm doing something wrong, or maybe my `.blerc` isn't being picked up, but nothing changes (I'm using it by `source ble.sh/out/ble.sh`).

**Reply by akinomyoga** [reply]
Thank you for checking. Hm, OK. Maybe I misunderstand the problem. Could you explain the problem you want to solve in a bit more detail?

I thought you would like to turn off the ble.sh feature `auto-complete` after you performed the TAB completion, but is "autocomplete" in the title of the question different from `auto-complete`? To make it clear, `auto-complete` and `complete` (the TAB completion) are two distinct features. I assumed that you would be talking about the interaction between these two features, but maybe you are actually talking about something different. For example, would you like to turn off the temporary insertion by `menu-complete` (though `menu-complete` is another feature distinct from `auto-complete` and the TAB completion)?

**Reply by DUOLabs333** [reply]
Ok, what I'm talking about is when you press TAB, a menu below showns up:
![image](https://user-images.githubusercontent.com/49244768/205327693-5ae88edb-65bb-4305-93a0-cf3821a1da19.png)
However, as you go through the menu, the choices replace whatever is in your current prompt:
![image](https://user-images.githubusercontent.com/49244768/205327866-2620b109-fe79-4c55-acaf-ff11aabb8f85.png)
I want to disable the choices replacing the prompt, but still allow me to select a choice, and only replacing the prompt when I press Enter on one of them.

**Reply by akinomyoga** [reply]
Thank you for your clarification! Ah, so I was completely missing what you mean. You are talking about the temporary insertion by `menu-complete` (which would be activated by another TAB after the TAB completion). I think I can also add an option for this. I'll later take a look.

**Comment by akinomyoga** [comment, marked as answer]
> However, as you go through the menu, the choices replace whatever is in your current prompt:
> ![image](https://user-images.githubusercontent.com/49244768/205327866-2620b109-fe79-4c55-acaf-ff11aabb8f85.png)
> I want to disable the choices replacing the prompt, but still allow me to select a choice, and only replacing the prompt when I press Enter on one of them.

Based on your feedback, I have newly added another option [`bleopt complete_menu_complete_opts`](https://github.com/akinomyoga/ble.sh/wiki/Manual-%C2%A77-Completion#user-content-bleopt-complete_menu_complete_opts) in commit 6a21ebb. To turn off the temporary insertion of the selected menu item in the commandline, you can use the following setting:

```bash
# blerc

bleopt complete_menu_complete_opts=
```

Could you update ble.sh by running `ble-update` and see if it is what you expect? Thank you!

**Reply by DUOLabs333** [reply]
It works perfectly. Thank you.

**Reply by akinomyoga** [reply]
Thanks for reporting!
